ch360/docslots_test.go

```diff
@@ -0,0 +1,86 @@
+package ch360_test
+
+import (
+	"context"
+	"errors"
+	"github.com/CloudHub360/ch360.go/ch360"
+	"github.com/CloudHub360/ch360.go/ch360/mocks"
+	"github.com/CloudHub360/ch360.go/test/generators"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"testing"
+)
+
+func Test_GetFreeDocSlots(t *testing.T) {
+	fixtures := []struct {
+		totalSlots    int
+		presentDocs   ch360.DocumentList
+		expectedSlots int
+		expectedErr   error
+		ctx           context.Context
+	}{
+		{
+			totalSlots:    10,
```

Comment:
I'm not really following what situations the different fixtures here are testing, I think partly because the numbers for each scenario seem to differ. Is it reasonable to expect that the number of expected slots is ever negative, for example?

Reply:
I'll add some comments to the test cases... but yes, right now, it is reasonable to assume that the expected slots is negative, since the number of total slots is currently hardcoded.

I've commented that behaviour in the `GetFreeDocSlots` function, just to be clear, and in that case (negative slots) the same error is returned as in the 0-slot case.